modules/h2store/examples/ExampleH2BrineSolubility.m

```diff
@@ -0,0 +1,280 @@
+% Rowe-Chou correlation. 
+
+%% Extract SAFT densities for H2 and H2O from state variables
+rhoH2_saft = state.DensH2; 
+rhoWater_saft = state.DensH2O;
+
+%% Constants
+mH2 = 2.016e-3;  % Molar mass of H2 (kg/mol)
+mWater = 0.0180153;  % Molar mass of H2O (kg/mol)
+Joule = 8.314472;  % Ideal gas constant (J/(mol*K))
+
+%% Extract RK-EoS table columns and convert units for comparison
+Temperature__K_ = tab_sol.("# temperature [°C]") + 273.15;  % Convert to Kelvin
+Pressure_Pa_ = tab_sol.("pressure [Pa]");
+
+%% Calculate density of hydrogen using the Brill and Beggs correlation
+ZH2 = calculateBrillBreggsZfactorHydrogen(Temperature__K_, Pressure_Pa_);
+rhoH2 = Pressure_Pa_ .* mH2 ./ (ZH2 .* Joule .* Temperature__K_);
+
+%% Calculate density of pure water using the Rowe-Chou correlation
+rhoWater = calculateRoweChouWaterDensity(Temperature__K_, 0);
+%% Plotting
+figure;
+hold on;
+% Loop over the four selected pressure levels
+for j = 1:4
+    i = kk(j);
+    % Convert the pressure value to MPa and set up legend label
+    P_val = unique(state.pressure(i:n:end));
+    P_legend = [num2str(int16(convertTo(P_val, mega * Pascal))) ' MPa'];
+    
+    % Plot ePC-SAFT results for H2 with solid lines
+    plot(Temperature__K_(i:n:n * n), rhoH2_saft(i:n:n * n, indexH2O), ...
+        'Color', plotColors{j}, 'LineStyle', '-', 'LineWidth', 1.5);
+    
+    % Plot NIST results for H2 with dashed lines for the same pressures
+    plot(Temperature__K_(i:n:n * n), tab_H2.("density [kg/m3]")(i:n:n * n), ...
+        'Color', plotColors{j}, 'LineStyle', '--', 'LineWidth', 1.5);
+    
+    % Plot Brill & Beggs correlation results for H2 with dash-dot lines
+    plot(Temperature__K_(i:n:n * n), rhoH2(i:n:n * n), ...
+        'Color', plotColors{j}, 'LineStyle', '-.', 'LineWidth', 2.0);
+    
+    % Set legend for each plot iteration
+    eval(['P' num2str(j) ' = [''ePC-SAFT ' P_legend ' MPa''];']);
+    eval(['R' num2str(j) ' = [''NIST ' P_legend ' MPa''];']);
+    eval(['S' num2str(j) ' = [''Corr. ' P_legend ' MPa''];']);
+end
+
+xlabel('Temperature (K)', 'FontSize', 14);
+ylabel('Density (kg/m³)', 'FontSize', 14);
+legend({P1, R1, S1, P2, R2, S2, P3, R3, S3, P4, R4, S4}, 'FontSize', 10, 'Location', 'best');
+title('H2 Density', 'FontSize', 12);
+hold off;
+%% Plotting Water Density
+figure;
+hold on;
+% Loop over the four selected pressure levels
+for j = 1:4
+    i = kk(j);
+    % Convert the pressure value to MPa and set up legend label
+    P_val = unique(state.pressure(i:n:end));
+    P_legend = [num2str(int16(convertTo(P_val, mega * Pascal))) ' MPa'];
+    
+    % Plot ePC-SAFT results for water density with solid lines
+    plot(Temperature__K_(i:n:n * n), rhoWater_saft(i:n:n * n), ...
+        'Color', plotColors{j}, 'LineStyle', '-', 'LineWidth', 1.5);
+    
+    % Plot NIST results for water density with dashed lines for the same pressures
+    plot(Temperature__K_(i:n:n * n), rhoWater(i:n:n * n), ...
```

Comment:
All the correlations and NIST data in this figure lie on top of each other. Are you supposed to be plotting tab_H2O here?

Reply:
A small nitpick is that examples usually are written with a small first letter since we reserve capital letters at the start for classes